pkg/interfaces/service.go

```diff
@@ -127,3 +133,10 @@ type DeviceServiceSDK interface {
 	// github.com/rcrowley/go-metrics
 	MetricsManager() interfaces.MetricsManager
 }
+
+// DeviceServiceSDKExt defines the extended contract for device sdk implementations that
+// provide additional APIs needed only from the sdk code.
+type DeviceServiceSDKExt interface {
```

Comment:
Why is this needed? Why can't Run() be part of the standard API above, Like it is in the App ADK.

Reply:
The app sdk expects the user to call `Run` directly. This sdk expects the user to call `startup.Bootstrap()`.